### PR TITLE
FEATURE: pagination for webhooks

### PR DESCRIPTION
**Pull request by fantasticfears**

### CODE REVIEW

File: app/controllers/admin/web_hooks_controller.rb [hunk 1]
```diff
@@ -2,9 +2,16 @@ class Admin::WebHooksController < Admin::AdminController
   before_filter :fetch_web_hook, only: %i(show update destroy)
 
   def index
-    # TODO: Pagination
+    limit = (params[:limit] || 50).to_i
```
Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: params[:limit].to_i || 50

File: app/controllers/admin/web_hooks_controller.rb [hunk 2]
```diff
@@ -2,9 +2,16 @@ class Admin::WebHooksController < Admin::AdminController
...
+    offset = params[:offset].to_i + params[:limit].to_i
```
Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: `limit` instead of `params[:limit]` ?

File: app/controllers/admin/web_hooks_controller.rb [hunk 3]
```diff
@@ -2,9 +2,16 @@ class Admin::WebHooksController < Admin::AdminController
...
+
+    total = WebHook.count
```
Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: Don't think we need to assign this to a variable

File: app/controllers/admin/web_hooks_controller.rb [hunk 4]
```diff
@@ -2,9 +2,16 @@ class Admin::WebHooksController < Admin::AdminController
...
+    web_hooks = WebHook.limit(limit).offset(offset)
```
Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: Same as https://github.com/discourse/discourse/pull/4331/files#r71067435

**Comment by fantasticfears** [comment]
I fixed some magic trick and eager-loads related associations. It should be a bit less bizarre :)

File: app/controllers/admin/web_hooks_controller.rb [hunk 5]
```diff
@@ -2,9 +2,18 @@ class Admin::WebHooksController < Admin::AdminController
   before_filter :fetch_web_hook, only: %i(show update destroy)
 
   def index
-    # TODO: Pagination
+    limit = params[:limit].to_i
+    limit = 50 if limit <= 0 || limit > 200
```
Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: hmmm why isn't the limitfixed?

Comment (fantasticfears) [inline app/controllers/admin/web_hooks_controller.rb]: Shall I make it an constant at 50? I can only imagine an API client would try a different number of `limit` so I set an upper bound here

Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: Yea just make it `50`

**Comment by tgxworld** [comment]
👍 Thanks

**Pull request merged**